util/FINETUNE_LM_PARAMETERS.md

```diff
@@ -0,0 +1,43 @@
+# Finetune Language Model Parameters: Alpha, Beta
+This topic is to explain how did `finetune_lm_params.py` fit `alpha` and `beta`
+
+## Inspiration
+According to mesh plot result of [Paddle/DeepSpeech-Grid-Search](https://github.com/PaddlePaddle/DeepSpeech/raw/develop/docs/images/tuning_error_surface.png), I assume `BatchWER v.s. Alpha` is `ideal valley parabolic curve`, i.e. which fits this formula below:
+
+Idea Valley Parabola Formula
+===
+- Assume there exists a `lowest WER` satisfies the formula: ![A.1](https://latex.codecogs.com/gif.latex?WER_{batch}=A_0(\\alpha-\\alpha_{best})^2+A_1 "A.1"),  `(A.1)`
+- Where ![](https://latex.codecogs.com/gif.latex?A_0>0) and ![](https://latex.codecogs.com/gif.latex?A_1>0)
+- If we set ![](https://latex.codecogs.com/gif.latex?\\alpha=\\alpha_{best}), the `WER` should be lowest: ![](https://latex.codecogs.com/gif.latex?WER_{lowest}=A_1), so the target is to find out the constant ![](https://latex.codecogs.com/gif.latex?\\alpha_{best})
+- To solve the `best alpha`, I decompose the formula, and change the coefficient as below:
+- ![A.2](https://latex.codecogs.com/gif.latex?WER_{batch}=A_0\\alpha^2-2A_0\\alpha_{best}\\alpha+A_1\\alpha_{best}^2+A_1 "A.2"),  `(A.2)`
+
+Polynomial Idea Valley Parabola Formula
+===
+- Let's tidy the `(A.2)` coefficients as below:
+- Assume there exists a `lowest WER` satisfies the formula: ![B.1](https://latex.codecogs.com/gif.latex?WER_{Batch}=B_0\\alpha^2+B_1\\alpha+B_2 "B.1"),  `(B.1)`
+- Where ![](https://latex.codecogs.com/gif.latex?B_0>0,B_1<0,B_2>0)
+- Compare to `(A.1)`: ![](https://latex.codecogs.com/gif.latex?B_0=A_0) and ![](https://latex.codecogs.com/gif.latex?B_1=-2A_0\\alpha_{best})
+- So, with formula `(B.1)`, we can derive: ![](https://latex.codecogs.com/gif.latex?\\alpha_{best}=-\\frac{B_1}{B_0})
+- To make computer calculations easier and consume the least amount of computing power, `Statistics` + `Linear Algebra` should be a good choice, so let's transform `(B.1)` as `matrix` form and introduce `statistics` assumption.
+
+
+Valley Parabola Formula in Matrix Form
+===
+- Assume testing one sampling audio with ![](https://latex.codecogs.com/gif.latex?\\alpha_n) by `test()` function can get ![](https://latex.codecogs.com/gif.latex?output_n), where `output` can be considered as `wer`/`cer`/`word_distance`/`char_distance`
+- Then if do an experiment on testing sampling data, with different `alpha` settings, we should get a bunch outputs: ![](https://latex.codecogs.com/gif.latex?\\textbf{Y}=\begin{bmatrix}output_0&output_1&\.\.\.&output_n\end{bmatrix})
+- To solve the best parameter issue by experiment data, we should introduce statistics method
```

Comment:
What statistics method? Sampling? Please fully describe what you're hinting at here.